### PR TITLE
ossm: add merge_upstream.sh to do merges of upstream commits

### PR DESCRIPTION
**Pull request by dgn**
Sadly, just merging commits isn't enough, we also need to re-generate some files. To not over-complicate the automator script, I put it into a separate script that we can call from the automator.

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -461,7 +461,7 @@ bundle: gen helm operator-sdk ## Generate bundle manifests and metadata, then va
 		if (git ls-files --error-unmatch "$$csvPath" &>/dev/null); then \
 			if ! (git diff "$$csvPath" | grep '^[+-][^+-][^+-]' | grep -v "createdAt:" >/dev/null); then \
 				echo "reverting timestamp change in $$csvPath"; \
-				git checkout "$$csvPath"; \
```
Comment (dgn) [inline Makefile.core.mk:464]: damn I'm just realizing I should make this change upstream. Submitting a PR...

File: ossm/merge_upstream.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+main () {  
+  if ! merge; then
+    set -e
+    echo "Conflicts detected, attempting to run 'make gen' to resolve."
+    rm -rf bundle resources
+    make gen
+    git add bundle resources chart
+    git -c "user.name=$GIT_USERNAME" -c "user.email=$GIT_EMAIL" commit --no-edit
```
Comment (FilipB) [inline ossm/merge_upstream.sh:71]: Do I understand this correctly? `git add`  will resolve conflicts and this commit will include also other not conflicting changes?

File: ossm/merge_upstream.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  GIT_EMAIL="$(curl -sSfLH "Authorization: token $GITHUB_TOKEN" "https://api.github.com/user" | jq --raw-output ".email")"
+fi
+
+GIT_COMMIT_MESSAGE=${GIT_COMMIT_MESSAGE:-"Automated merge"}
+
+MERGE_STRATEGY=${MERGE_STRATEGY:-"merge"}
+MERGE_REPOSITORY=${MERGE_REPOSITORY:-"git@github.com:istio-ecosystem/sail-operator.git"}
+MERGE_BRANCH=${MERGE_BRANCH:-"main"}
+
+print_error() {
```
[the remark this answers is not in the document]
Comment (dgn) [reply ossm/merge_upstream.sh:39]: oh, whoops. yeah, I copied a bunch of code from the automator script that required this, but I have refactored a few times and don't actually need it anymore

File: ossm/merge_upstream.sh (hunk 2 above)
Comment (dgn) [reply ossm/merge_upstream.sh:71]: `rm` and `make gen` resolve the conflicts, `git add` marks them as resolved and `git commit` finishes the merge, correct

Comment (dgn) [reply ossm/merge_upstream.sh:59]: the commit is basically like `git merge --continue` but non-interactive

Comment (FilipB) [reply ossm/merge_upstream.sh:59]: Thanks!

**Review by FilipB — APPROVED** [review]
LGTM

**Pull request merged**